source/auth/tests/legacy/connection-string.json

```
@@ -569,6 +569,36 @@
         }
       }
     },
+    {
+      "description": "should accept a url-encoded TOKEN_RESOURCE (MONGODB-OIDC)",
+      "uri": "mongodb://user@localhost/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:mongodb%253A//test-cluster",
+      "valid": true,
+      "credential": {
+        "username": "user",
+        "password": null,
+        "source": "$external",
+        "mechanism": "MONGODB-OIDC",
+        "mechanism_properties": {
+          "ENVIRONMENT": "azure",
+          "TOKEN_RESOURCE": "mongodb%253A//test-cluster"
```

This appears to be double-encoded - I do not think that `mongodb%3A//test-cluster` is the intended final value of the resource. It is fine that this is testing a partial encoding, where unlike in `mongodb%3A%2F%2Ftest-cluster`, the `//` is not encoded but the `:` is, but a single decode would be:
```suggestion
          "TOKEN_RESOURCE": "mongodb%3A//test-cluster"
```

---

It seems users MUST encode property values, and then MUST encode at least the TOKEN_RESOURCE sub-value. This is unusual and unintuitive, because let’s say the user wants to use an already-url-encoded value, like `something?hello%20world` for the resource. Great - just pass it in as a url-encoded value `something%3Fhello%2520world` right after TOKEN_RESOURCE:
```
&authMechanismProperties=ENVIRONMENT:azure,TOKEN_RESOURCE:something%3Fhello%2520world
```
But then, it gets decoded once by the parameter decoder:
```
ENVIRONMENT:azure,TOKEN_RESOURCE:something?hello%20world
```
And then again by the TOKEN_RESOURCE value decoder:
```
something?hello world
```
The `%20` was lost. The only solutions seem to be to a) document that users MUST url-encode parameters when providing a url-encoded value, and we can’t just say “double encode the resource value”, and all of this is confusing, or b) to not encode TOKEN_RESOURCE using url-encoding, but some other encoding that won’t be decoded by the param decoder.